snabbdom/src/main/scala/snabbdom/AttrValue.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022 buntec
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copyright (c) 2015 Simon Friis Vindum
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+package snabbdom
+
+sealed trait AttrValue
+
+object AttrValue {
+
+  def apply(s: String): AttrValue = StringAttrValue(s)
+  def apply(b: Boolean): AttrValue = BooleanAttrValue(b)
+
+  case class StringAttrValue(value: String) extends AttrValue
+  case class BooleanAttrValue(value: Boolean) extends AttrValue
+
+  implicit def stringToAttrValue(value: String): AttrValue =
```

Comment:
I know this is controversial but still seems to be quite common for such cases, see e.g., [here](https://github.com/tpolecat/natchez/blob/series/0.1/modules/core/shared/src/main/scala/TraceValue.scala)

Reply:
Yeah. I think it goes to @cornerman's question in https://github.com/buntec/scala-js-snabbdom/issues/14#issuecomment-1120278952

> Do we consider this library directly user facing or is it more like a low level library powering, e.g., ff4s and outwatch.

If it's a low-level library, we probably wouldn't need this, right?